source/io/module_javaScriptInvoke.js

```diff
@@ -505,6 +533,22 @@ var assignJavaScriptInvoke;
             var isNotAJavaScriptRefnum = !hasCachedRefNum(cookie);
             Module.eggShell.writeDouble(returnValueRef, isNotAJavaScriptRefnum ? 1 : 0);
         };
+
+        /**
+         * Static references (ie, control reference) aren't closed and not removed from the map
+         */
+        Module.javaScriptInvoke.jsCloseJavaScriptRefNum = function (javaScriptRefnumTypeRef, javaScriptRefnumDataRef, errorTypeRef, errorDataRef) {
+            var javaScriptValueRef = Module.eggShell.createValueRef(javaScriptRefnumTypeRef, javaScriptRefnumDataRef);
+            var isDynamicReference = Module.typeHelpers.isJSObjectDynamicRefnum(javaScriptValueRef.typeRef);
+            if (isDynamicReference) {
+                var cookie = Module.eggShell.readDouble(javaScriptValueRef);
+                var keyWasPresent = cookieToJsValueMap.delete(cookie);
+                if (!keyWasPresent) {
+                    var errorValueRef = Module.eggShell.createValueRef(errorTypeRef, errorDataRef);
+                    mergeNewError(errorValueRef, 'CloseReference', ERRORS.kNIInvalidReference);
```

Comment:
This was more correct before. The mergeNewError helper function name is the JS function name, not the VI name.

Close should not use this helper

Reply:
Done. Reverted to previous implementation.